mijia/src/bluetooth.rs

```diff
@@ -114,6 +178,25 @@ pub struct DeviceInfo {
     pub service_data: HashMap<String, Vec<u8>>,
 }
 
+/// Information about a GATT service on a Bluetooth device.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct ServiceInfo {
+    /// An opaque identifier for the service on the device, including a reference to which adapter
+    /// it was discovered on.
+    pub id: ServiceId,
+    pub uuid: String,
+    pub primary: bool,
+}
+
+/// Information about a GATT characteristic on a Bluetooth device.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct CharacteristicInfo {
+    /// An opaque identifier for the characteristic on the device, including a reference to which
+    /// adapter it was discovered on.
+    pub id: CharacteristicId,
+    pub uuid: String,
```

Comment:
there is a uuid crate. Could we type this as
```
struct CharacteristicUuid(uuid::Uuid)
```
?

Reply:
I'm not really keen on making a new type, as it really is just a UUID, but I've used the `uuid` crate.